sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -158,15 +332,21 @@ static ProcessElementMethod create(
     }
 
     /** Whether this {@link DoFn} uses a Single Window. */
-    public boolean usesSingleWindow() {
-      return extraParameters().contains(Parameter.BOUNDED_WINDOW);
+    public boolean ignoresWindow() {
+      for (Parameter p : extraParameters()) {
+        if (p instanceof BoundedWindowParameter
+            || p instanceof StateParameter) {
```

Comment:
May be worth a comment about why having a StateParameter constitutes not ignoring windows?

Also, may be worth:
```
return !Iterables.any(extraParameters(), Predicates.or(
    Predicates.instanceOf(BoundedWindowParameter.class),
    Predicates.instanceOf(StateParameter.class)));
```